### PR TITLE
Feature to add swaps to an address instead of a wallet

### PR DESCRIPTION
**Pull request by markettes**
Wait for the merge, since Liquidator and other components depend on this.

### CODE REVIEW

File: src/Rpc/NodeGuardService.cs [hunk 1]
```diff
@@ -693,18 +694,47 @@ public override async Task<AddLiquidityRuleResponse> AddLiquidityRule(AddLiquidi
...
+        var swapWallet = await _walletRepository.GetById(request.SwapWalletId);
+        if (swapWallet == null)
         {
-            throw new RpcException(new Status(StatusCode.NotFound, "Wallet not found"));
+            throw new RpcException(new Status(StatusCode.NotFound, "Swap wallet not found"));
+        }
+
+        if (request.IsReverseSwapWalletRule)
+        {
+            if (!request.HasReverseSwapWalletId)
```
Comment (RodriFS) [inline src/Rpc/NodeGuardService.cs:705]: why is this necessary? maybe is better to compare `request.ReverseSwapWalletId == null` and have one less field that might not be updated

File: src/Rpc/NodeGuardService.cs [hunk 2]
```diff
@@ -693,18 +694,47 @@ public override async Task<AddLiquidityRuleResponse> AddLiquidityRule(AddLiquidi
...
+                throw new RpcException(new Status(StatusCode.FailedPrecondition, "WalletId is required for wallet rules"));
+            
+            var wallet = await _walletRepository.GetById(request.ReverseSwapWalletId);
+            if (wallet == null)
+                throw new RpcException(new Status(StatusCode.NotFound, "Wallet not found"));
+        }
+        else
+        {
+            if (!request.HasReverseSwapAddress)
```
Comment (RodriFS) [inline src/Rpc/NodeGuardService.cs:714]: same here

**Review by RodriFS — APPROVED** [review]

File: src/Rpc/NodeGuardService.cs (hunk 1 above)
Comment (markettes) [reply src/Rpc/NodeGuardService.cs:705]: This is more of a double check, if someone forgot to remove when changing and both fields were populated this field would be essential to determine the way to execute the swap.

Comment (RodriFS) [reply src/Rpc/NodeGuardService.cs:705]: I meant that the field seems unnecessary, not the check. But is just to be nitpicky

File: src/Data/ApplicationDbContext.cs [hunk 3]
```diff
@@ -43,13 +43,21 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
                 .WithMany(node => node.ChannelOperationRequestsAsDestination)
                 .HasForeignKey(cor => cor.DestNodeId);
+            modelBuilder.Entity<LiquidityRule>()
+                .HasOne(lr => lr.SwapWallet)
+                .WithMany(wallet => wallet.LiquidityRulesAsSwapWallet)
+                .HasForeignKey(lr => lr.SwapWalletId);
+            modelBuilder.Entity<LiquidityRule>()
+                .HasOne(lr => lr.ReverseSwapWallet)
+                .WithMany(wallet => wallet.LiquidityRulesAsReverseSwapWallet)
+                .HasForeignKey(lr => lr.ReverseSwapWalletId);
```
Comment (Jossec101) [inline src/Data/ApplicationDbContext.cs:60]: NIT: Using the fluent api for one-to-many is not strictly needed.

File: src/Pages/Channels.razor [hunk 4]
```diff
@@ -303,23 +303,66 @@
...
+                                    <Validation Validator="ValidationRule.IsSelected">
+                                        <FieldLabel>Wallet to use in SWAPS operations</FieldLabel>
+                                        <SelectList TItem="Wallet"
+                                                    TValue="int"
+                                                    Data="@_availableWallets"
+                                                    SelectedValue="_currentLiquidityRule.SwapWalletId"
+                                                    TextField="@((item) => item.Name)"
+                                                    ValueField="@((item) => item.Id)"
+                                                    SelectedValueChanged="@OnSelectedSwapWalletChanged"
+                                                    DefaultItemText="Choose the wallet to fund/receive swap amounts">
```
Comment (Jossec101) [inline src/Pages/Channels.razor:319]:
```suggestion
                                                    DefaultItemText="Choose the wallet to fund or receive the swaps">
```

**Review by Jossec101 — APPROVED** [review]

File: src/Data/ApplicationDbContext.cs (hunk 3 above)
Comment (markettes) [reply src/Data/ApplicationDbContext.cs:60]: It was asking me for it

**Review by RodriFS — APPROVED** [review]
